tpm2_pytss/types.py

```diff
@@ -732,399 +743,715 @@ class TPMA_MEMORY(int):
+    def __getitem__(self, item):
+        try:
+            return object.__getitem__(self, item)
+        except AttributeError:
+            if isinstance(item, slice):
+                if item.stop is None:
+                    item = slice(item.start, len(self) - 1, item.step)
+                digest_list = self._cdata.digests[item]
+                return [TPM2B_unpack(x) for x in digest_list]
 
-class TPM2B_PRIVATE(object):
-    def __new__(self):
-        return ffi.new("TPM2B_PRIVATE *")
+            return TPM2B_unpack(self._cdata.digests[item])
 
 
-class TPM2B_PRIVATE_KEY_RSA(object):
-    def __new__(self):
-        return ffi.new("TPM2B_PRIVATE_KEY_RSA *")
+class TPML_DIGEST_VALUES(TPML_OBJECT):
+    def __init__(self, _cdata=None):
+        init_data = TPM_OBJECT._init_cdata(self, _cdata)
+        super().__init__(init_data)
 
 
-class TPM2B_PRIVATE_VENDOR_SPECIFIC(object):
-    def __new__(self):
-        return ffi.new("TPM2B_PRIVATE_VENDOR_SPECIFIC *")
+class TPML_ECC_CURVE(TPML_OBJECT):
+    def __init__(self, _cdata=None):
+        init_data = TPM_OBJECT._init_cdata(self, _cdata)
+        super().__init__(init_data)
 
 
-class TPM2B_PUBLIC(object):
-    def __new__(self):
-        return ffi.new("TPM2B_PUBLIC *")
+class TPML_HANDLE(TPML_OBJECT):
+    def __init__(self, _cdata=None):
+        init_data = TPM_OBJECT._init_cdata(self, _cdata)
+        super().__init__(init_data)
 
 
-class TPM2B_PUBLIC_KEY_RSA(object):
-    def __new__(self):
-        return ffi.new("TPM2B_PUBLIC_KEY_RSA *")
+class TPML_INTEL_PTT_PROPERTY(TPML_OBJECT):
+    def __init__(self, _cdata=None):
+        init_data = TPM_OBJECT._init_cdata(self, _cdata)
+        super().__init__(init_data)
 
 
-class TPM2B_SENSITIVE(object):
-    def __new__(self):
-        return ffi.new("TPM2B_SENSITIVE *")
+class TPML_PCR_SELECTION(TPML_OBJECT):
+    def __init__(self, pcr_selections=None, _cdata=None):
+        init_data = TPM_OBJECT._init_cdata(self, _cdata)
+        super().__init__(init_data)
 
+        if pcr_selections is None or len(pcr_selections) == 0:
+            return
 
-class TPM2B_SENSITIVE_CREATE(object):
-    def __new__(self):
-        return ffi.new("TPM2B_SENSITIVE_CREATE *")
+        if len(pcr_selections) > lib.TPM2_NUM_PCR_BANKS:
+            raise RuntimeError(
+                f"PCR Selection list greater than {lib.TPM2_NUM_PCR_BANKS}, "
+                f"got {len(pcr_selections)}"
+            )
 
+        self._cdata.count = len(pcr_selections)
+        for i, p in enumerate(pcr_selections):
+            self._cdata.pcrSelections[i] = p._cdata[0]
 
-class TPM2B_SENSITIVE_DATA(object):
-    def __new__(self):
-        return ffi.new("TPM2B_SENSITIVE_DATA *")
+    @staticmethod
+    def parse(selections):
 
+        if selections is None or len(selections) == 0:
+            return TPML_PCR_SELECTION()
 
-class TPM2B_SYM_KEY(object):
-    def __new__(self):
-        return ffi.new("TPM2B_SYM_KEY *")
+        selectors = selections.split("+")
```

Comment:
Similar to other comment, You might want to check for the presence of "+" before splitting on it, and similarly raise an exception to the one below in the bounds checking. Users might get an unhelpful error message if they didn't supply a value with a "+".

Reply:
Theirs a chance only one bank is specified and no + between them, so we want to make sure we handle and test that case. Good catch.